### PR TITLE
Create CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by jhamman**
This adds a code of conduct document to the top level of the zarr repo. Based on discussion in #291.

There is one todo item. We need to find an email address to route reports of unacceptable behavior, should they occur (see line 37 in the document).

### CODE REVIEW

**Comment by alimanfoo** [comment]
BTW do we need to include a code of conduct in the numcodecs repo also?

**Comment by jhamman** [comment]
I think it would be fine to duplicate this text in each zarr-developers repo. +1 on a dedicated gmail account.

**Comment by jakirkham** [comment]
> BTW do we need to include a code of conduct in the numcodecs repo also?

I guess the alternative would be to add it to a central repo (webpage repo?) and link to it from other contribution docs. Depends what organizational structure we like and how frequently we plan to update this.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Scope
+
+This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at TODO@something.com. The project team will review and investigate all complaints, and will respond in a way that it deems appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
```
Comment (alimanfoo) [inline CODE_OF_CONDUCT.md]: I've created zarr.conduct@gmail.com.

**Comment by alimanfoo** [comment]
FWIW I think it would be easiest to duplicate this in the numcodecs repo for now. We can always consolidate later.

**Comment by jhamman** [comment]
Thanks @alimanfoo - I updated the doc accordingly.

**Comment by alimanfoo** [comment]
Thanks @jhamman.

**Pull request merged**